### DISCUSSION TITLE
getting sqlite returning to work

### DISCUSSION THREAD

**Discussion by mathause** [sqlalchemy/sqlalchemy#9556] [Usage Questions] [answered]
I have problems getting `RETURNING` to work in sqlite with sqlalchemy 2.0.7 (with core - haven't tested ORM):

```python
import sqlalchemy

metadata = sqlalchemy.MetaData()

url = 'sqlite:///test.db'

engine = sqlalchemy.create_engine(url, echo=True)

notes = sqlalchemy.Table(
    "notes",
    metadata,
    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
    sqlalchemy.Column("text", sqlalchemy.String),
)

metadata.create_all(engine)

query = sqlalchemy.insert(notes).values(text="NOTE").returning(notes.c.id)

with engine.begin() as conn:
    result = conn.execute(query)
```

This returns the following error:

```python-traceback
OperationalError: (sqlite3.OperationalError) cannot commit transaction - SQL statements in progress
(Background on this error at: https://sqlalche.me/e/20/e3q8)
```

I can get `RETURNING` to work in pure sqlite3 and according to #6195 this should work in sqlalchemy 2.0. What am I missing? (The example works with postgesql. I also could not find an example where a `RETURNING` query is actually committed in the docs.)

<details>

```python

import sqlite3
con = sqlite3.connect('example.db')

con.execute("CREATE TABLE test (id INTEGER PRIMARY KEY, text TEXT );")

result = con.execute("INSERT INTO test (text) VALUES ('TEXT') RETURNING test.id")

result.fetchall()
```

</details>

**Comment by CaselIT** [comment, marked as answer]
Hi,

Try calling `result.all()` to fetch the rows that are returned from the statement.
The error seems to indicate that commit cannot happen because the returning statement has not been fetched

**Reply by mathause** [reply]
And it's so easy! Thanks - somehow I couldn't figure out how to make this work.

**Reply by CaselIT** [reply]
my guess was the message, and the fact that the example that worked used `result.fetchall()` while in the sqlalchemy version it was missing

glad it helped
